actionpack/lib/abstract_controller/caching/fragments.rb

```diff
@@ -136,7 +136,7 @@ def expire_fragment(key, options = nil)
 
       def instrument_fragment_cache(name, key) # :nodoc:
         payload = instrument_payload(key)
-        ActiveSupport::Notifications.instrument("#{name}.#{instrument_name}", payload) { yield }
+        ActiveSupport::Notifications.instrument("#{name}.#{instrument_name}".freeze, payload) { yield }
```

Comment:
hey FWIW cached frozen strings don't work like that when the string is interpolated. Adding `.freeze` on this line will still allocate new objects.

**what's happening here:**
```ruby
    GC.start
    before = GC.stat(:total_allocated_objects)

    this_is_static = "hello".freeze
    100_000.times do |i|
      "#{this_is_static} this is also static".freeze
    end

    GC.start
    after = GC.stat(:total_allocated_objects)
    puts "Objects added: #{after - before}" #=> 100000
```

**what was happening in the other cases:**
```ruby
    GC.start
    before = GC.stat(:total_allocated_objects)

    100_000.times do |i|
      "this is static".freeze
    end

    GC.start
    after = GC.stat(:total_allocated_objects)
    puts "Objects added: #{after - before}" #=> 0
```